vcpkg/maintainers/publishing-to-a-git-registry.md

```diff
@@ -0,0 +1,124 @@
+
+Finally, let's commit these changes to the repository.
+
+```Console
+git add .
+git commit -m "Initial commit"
+```
+
+Optionally, if you have a remote server, this is a good point to commit your changes.
+
+```Console
+git remote add origin https://example.com/vcpkg-registry.git
+git push --set-upstream origin HEAD
+```
+
+## Step 3: Add ports to the registry
+
+In this step you should add all your ports in the `ports` folder. Each port must have.
+its own subdirectory containing at least a `vcpkg.json` file and a `portfile.cmake` file.
+Learn more about creating ports in the tutorial on [packaging GitHub
+repositories](../examples/packaging-github-repos.md).
+
+When all your ports have been added commit your changes to the repository.
+
+```Git
+git add ports/.
+git commit -m "Add registry ports"
+```
+
+## Step 4: Update the versions database
+
+> [!NOTE]
+> This step assumes that you're familiar with some versioning concepts.
+> Read the [versioning documentation](../users/versioning.md) to learn about the
+> versions database and baseline file.
+
+Let's assume that you have the following ports in the `ports` directory.
+
+```Console
+ls ports
+beicode/ fmt/
+```
+
+The next step is to generate the version database files for your ports and
+also update the `baseline.json` file. To do that, let's leverage the
+`vcpkg x-add-version` command.
+
+From the root of your registry run the following:
+
+```Console
+vcpkg --x-builtin-ports-root=./ports --x-builtin-registry-versions-dir=./versions x-add-version --all --verbose
+```
+
+Which produces the output:
+
+```Console
+added version 1.0.0 to /home/vcpkg/vcpkg-registry/versions/b-/beicode.json (new file)
+added version 1.0.0 to /home/vcpkg/vcpkg-registry/versions/baseline.json
+added version 10.0.0 to /home/vcpkg/vcpkg-registry/versions/f-/fmt.json (new file)
+added version 10.0.0 to /home/vcpkg/vcpkg-registry/versions/baseline.json
+```
+
+The `x-add-version` command updates the versions database,
+it is normally used for the official vcpkg registry in <https://github.com/Microsoft/vcpkg>.
+But by adding some experimental options it can be made to work on any Git-based registry.
+
+It is important that all pending changes for your ports have been comitted to the registry's
+repository before running the command.
+
+The `--x-builtin-ports-root=./ports` and `--x-builtin-registry-versions-dir=./versions`, redirect
+vcpkg to treat your registry's `ports` and `versions` folders as if they were the built-in ones (by
+built-in we refer to the ports registry that ships with the vcpkg repository). The `--all` options
+indicates that the command should scan the `ports` folder for all the ports it can find, and the
+`--verbose` option makes it so that the command produces output back to the terminal.
+
+The result of this command is that version database files for each port are created or updated as
+necessary, at the same time the `versions/baseline.json` file is updated with the newly added
+versions.
```

Comment:
This may seem obvious, but it might help to show what is the expected outcome. Can the user now use these ports in a vcpkg workflow? Maybe some sort of example workflow would be useful here.

Reply:
There is a separate article for "Consuming packages from a registry", this article should link to it eventually.